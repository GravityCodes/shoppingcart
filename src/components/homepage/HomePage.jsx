import styles from "./HomePage.module.css"
import heroImg from "../../assets/imgs/hero-background.jpg"
import { Link } from "react-router-dom"
import { useCategories } from "../categories/useCategories"
import { CategoryButton } from "../categories/CategoryButton"
import { FeatureProducts } from "../featureProducts/FeatureProducts"
import {Header} from "../header/Header"
import { useCartProducts } from "../../hooks/useCartProducts"

export const HomePage = () => {

  const {categories, error, loading} = useCategories();
  const { updateCartProducts, amountOfProducts} = useCartProducts();

  return (
    <>
      <Header numOfProducts={amountOfProducts} />
    
      <div className={styles.hero}>
        <img src={heroImg} alt="" />
        <p className={styles["hero-children"]}>Reasonable products for Reasonable prices</p>
        <div className={styles["hero-children-link-wrapper"]}>
        <Link className={styles["link-btn"]} to={"/shop"}> Start Shopping</Link>
        </div>
      </div>

      <div className={styles["categories-button-grid"]}>
        {
          loading ? <p>Loading...</p> 
          : error ? <p> A network Error has been encountered</p>
          : categories.map(category => <CategoryButton key={category} name={category} />)
        }
      </div>
      
      <FeatureProducts updateCartProducts={updateCartProducts} />
    
    </>
  )
}