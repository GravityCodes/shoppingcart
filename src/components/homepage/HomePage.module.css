h1 {
  font-size: 2rem;
  text-align: center;
}

.hero {
  position: relative;
  height: calc(100vh - 50px);
}

.img {
  height: 100%;
  width: 100%;
  object-fit: cover;
}

.hero-children {
  position: absolute;
  top: 40%;
  color: white;
  width: 100%;
  text-align: center;
  font-size: 1.5rem;
}

.hero-children-link-wrapper {
  position: absolute;
  top: 50%;
  width: 100%;
  text-align: center;
}

.box {
  width: 100%;
  backdrop-filter: blur(3px);
}

.link-btn {
  border: 2px solid white;
  padding: 6px;
  color: white;
  text-decoration: none;
  position: relative;
  perspective: 400px;
  transform: translateZ(2px);
}

.link-btn::before {
  content: "";
  position: absolute;
  z-index: -1;
  height: 100%;
  width: 100%;
  backdrop-filter: blur(5px);
  transform: translate(-5px, -3px) translateZ(1px);
}

.categories-button-grid {
  display: flex;
  flex-wrap: wrap;
  gap: 20px;
  white-space: nowrap ;
  justify-content: center;
  padding: 20px;
}

.category-loading {
  height: 50px;
  width: 50px;
  border-radius: 50%;
  background-color: rgb(196, 196, 196);
  opacity: 1;
  margin: 0 auto;
  animation: pulse 0.7s infinite alternate;
}

@keyframes pulse {
  to {
    opacity: 0.3;
  }
}

@media (min-width: 426px) {
  .hero-children {
    font-size: 1.2rem;
  }
  
}

@media (min-width: 849px) {
  .hero-children {
    font-size: 1.5rem;
  }
}