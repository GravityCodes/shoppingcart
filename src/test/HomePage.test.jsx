import { describe, it, expect } from "vitest"
import {render} from "@testing-library/react"
import { BrowserRouter } from "react-router-dom"
import {HomePage} from "../components/homepage/HomePage"

describe("HomePage component", () => {
  it("matches snapshot", () => {
   const {container} = render(
      <BrowserRouter>
        <HomePage />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  })
})