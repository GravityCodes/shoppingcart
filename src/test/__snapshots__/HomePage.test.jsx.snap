// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage component > matches snapshot 1`] = `
<div>
  <div
    class="_header_69e643"
  >
    <button>
       
      <svg
        class="lucide lucide-menu _menu_26077f"
        fill="none"
        height="22"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="4"
          x2="20"
          y1="12"
          y2="12"
        />
        <line
          x1="4"
          x2="20"
          y1="6"
          y2="6"
        />
        <line
          x1="4"
          x2="20"
          y1="18"
          y2="18"
        />
      </svg>
       
    </button>
    <h1
      role="heading"
    >
      <a
        data-discover="true"
        href="/"
      >
        Fordly
      </a>
    </h1>
    <a
      data-discover="true"
      href="/Checkout"
    >
      <div
        class="_cart_6d819b"
      >
        <svg
          class="lucide lucide-shopping-cart"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="8"
            cy="21"
            r="1"
          />
          <circle
            cx="19"
            cy="21"
            r="1"
          />
          <path
            d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
          />
        </svg>
      </div>
    </a>
  </div>
  <div
    class="_hero_d8ed07"
  >
    <img
      alt=""
      class="_img_d8ed07"
      src="/src/assets/imgs/hero-background.jpg"
    />
    <p
      class="_hero-children_d8ed07"
    >
      Reasonable products for Reasonable prices
    </p>
    <div
      class="_hero-children-link-wrapper_d8ed07"
    >
      <a
        class="_link-btn_d8ed07"
        data-discover="true"
        href="/shop"
      >
         Start Shopping
      </a>
    </div>
  </div>
</div>
`;
